### DISCUSSION TITLE
dokku-postgres: set DATABASE_URL with different name and without _URL in the end

### DISCUSSION THREAD

**Discussion by ohld** [dokku/dokku#4370] [Q&A]
Hey 👋

I'm not sure that this question should be asked here but I'll try.

I want to deploy production Airflow using Dokku. I want to attach Postgres plugin to the app using `AIRFLOW__CORE__SQL_ALCHEMY_CONN` env variable ([link to Airflow tutorial](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html#setting-configuration-options)). When I try to use `--alias` param in `dokku postgres:link` command, it links database using `AIRFLOW__CORE__SQL_ALCHEMY_CONN_URL` env variable. So it adds *_URL* to the end. But in that case Airflow can't see the database!

How can I link Postgres database using the proper env var name? I can set it manually using `dokku config:set` but I'm not sure that this is the right way for production. Thanks!

**Comment by Cellane** [comment]
I’m somehow afraid that the plugin always appends the `_URL` to the environmental variable name, on [this line probably](https://github.com/dokku/dokku-postgres/blob/e414eaccec7c1ec5af603fc03e95c696d4d77dce/common-functions#L468).

I wonder if that should be the case, it would make sense to me to append `_URL` in case of default alias, but omit it if custom alias is provided…

**Reply by ohld** [reply]
I would be great if I count somehow specify a full final env name without forced suffix `_URL`.

**Comment by josegonzalez** [comment]
The alias always has the suffix appended. Dropping that would change behavior and would be a major breaking change. I would argue that airflow is doing something weird here with their url handling. The majority of tools I've used which auto-inhale env vars that also don't use `_URL` as the suffix.

**Reply by ohld** [reply]
I agree that the majority of tools may be using _URL prefix but I believe that the most scalable solution is to have the ability to set the database url without this prefix.

**Reply by ohld** [reply]
For example, I faced the same problem while deploying my beloved open source BI tool Metabase. For production, it requires database url to be named `MB_DB_CONNECTION_URI` ([source](https://www.metabase.com/docs/latest/operations-guide/configuring-application-database.html)). Which doesn't have `_URL` suffix as well.

**Reply by josegonzalez** [reply]
I don't know that "scalable" is the right work - perhaps "flexible" is better.

Anyhow, I wonder if we could add support for another name for alias to allow folks to override the env var completely? Whats an alias for `alias`?

**Comment by ohld** [comment]
I tried to come up with a short script that will allow to easily run Metabase on Dokku. I did it but I can't just copy-paste it to shell because I need to manually insert database url using `dokku config:set ...`.

This is my Gist. On the 12th line you can find what I'm talking about.
https://gist.github.com/ohld/aa9bb626e1abea5310fa9ff1094ec16e

So it would be great to have a way to list Postgres db to a Dokku app using any env variable name without mandatory `_URL` suffix.

**Reply by nickhammond** [reply]
From looking at your gist, what about just fetching the config var from the official PG link and set it that way? Something like:

```
dokku config:set metabase MB_DB_TYPE=postgres MB_DB_CONNECTION_URI=$(dokku config:get metabase DATABASE_URL)
```
